framework/console/Application.php

```diff
@@ -165,20 +165,20 @@ public function handleRequest($request)
      * code should be used:
      *
      * ```php
-     * \Yii::$app->runAction('controller/test', ['option' => 'value', $a, $b]);
+     * \Yii::$app->runAction(Yii::$app->getRequest(), 'controller/test', ['option' => 'value', $a, $b]);
      * ```
      *
+     * @param Request $request the request instance.
      * @param string $route the route that specifies the action.
      * @param array $params the parameters to be passed to the action
      * @return int|Response the result of the action. This can be either an exit code or Response object.
-     * Exit code 0 means normal, and other values mean abnormal. Exit code of `null` is treaded as `0` as well.
+     * Exit code 0 means normal, and other values mean abnormal. Exit code of `null` is treated as `0` as well.
      * @throws Exception if the route is invalid
      */
-    public function runAction($route, $params = [])
+    public function runAction($request, $route, $params = [])
```

Comment:
Could be type-hinted with `Request`.

Reply:
Using `yii\console\Request` as a type restriction is a risk prone. Unless there is some `yii\console\RequestInterface` interface type-restriction introduction reduces flexibility here.

Reply:
OK.